### PR TITLE
chore(react): improve react e2e tests speed

### PR DESCRIPTION
**Pull request by xiongemi**
## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by mandarini — APPROVED** [review]

File: e2e/react-core/src/react-module-federation.test.ts
Comment (AgentEnder) [inline e2e/react-core/src/react-module-federation.test.ts:1]: Can you update the CODEOWNERs files for the new paths?

**Review by AgentEnder — CHANGES_REQUESTED** [review]

File: e2e/react-extensions/jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable */
+export default {
+  transform: {
+    '^.+\\.[tj]sx?$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
+  },
+  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
+  maxWorkers: 1,
+  globals: {},
+  displayName: 'e2e-react',
```
Comment (barbados-clemens) [inline e2e/react-extensions/jest.config.ts:9]:
```suggestion
  displayName: 'e2e-react-extensions',
```

**Review by barbados-clemens — COMMENTED** [review]
just a small name change so we know which projects we are looking at in the logs.

File: e2e/react-extensions/jest.config.ts (hunk 1 above)
Comment (xiongemi) [reply e2e/react-extensions/jest.config.ts]: changed

File: e2e/react-core/src/react-module-federation.test.ts
Comment (xiongemi) [reply e2e/react-core/src/react-module-federation.test.ts:1]: in the CODEOWNERS, it points to entire /e2e/**/* folder. but I did update the codeowners in .github/workflows/e2e-matrix.yml

Comment (AgentEnder) [reply e2e/react-core/src/react-module-federation.test.ts:1]: That's just a fallback - under the react vertical there are specific entries for e2e that need to be updated.

Comment (AgentEnder) [reply e2e/react-core/src/react-module-federation.test.ts:1]: This line:
```
/e2e/react/** @jaysoo @mandarini @xiongemi @ndcunningham
```
Should change to something like:
```
/e2e/react*/** @jaysoo @mandarini @xiongemi @ndcunningham
```

**Review by barbados-clemens — APPROVED** [review]

**Review by meeroslav — APPROVED** [review]
I have rebased the repo, so the issue with `e2e-react` should no longer appear.

Unfortunately latest master also brings another bug fixed in https://github.com/nrwl/nx/pull/16152

**Review by jaysoo — APPROVED** [review]

File: e2e/react-core/src/react-module-federation.test.ts
Comment (xiongemi) [reply e2e/react-core/src/react-module-federation.test.ts:1]: changed.

**Review by AgentEnder — APPROVED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**